lib/jekyll-admin/server.rb

```diff
@@ -75,7 +75,11 @@ def write_file(path, content)
       File.open(path, "wb") do |file|
         file.write(content)
       end
-      site.process
+      if ENV["RACK_ENV"]
+        site.process
+      else
+        site.read
+      end
```

Comment:
@parkr See https://github.com/jekyll/jekyll-admin/issues/289#issuecomment-278365937 for context. Is the race condition in the write? In the read? In the generate? Any way to unthread things? We need to write and generate so previews work.